Sources/Basics/FileSystem+Extensions.swift

```diff
@@ -110,3 +110,31 @@ extension FileSystem {
         return idiomaticConfigDirectory
     }
 }
+
+// MARK: - script cache
+
+extension FileSystem {
+    /// SwiftPM cache directory under user's caches directory (if exists)
+    public var swiftScriptCacheDirectory: AbsolutePath {
+        return self.dotSwiftScriptCachesDirectory
+    }
+
+    fileprivate var dotSwiftScriptCachesDirectory: AbsolutePath {
+        return self.dotSwiftPM.appending(component: "scripts")
+    }
+}
+
+extension FileSystem {
+    public func getOrCreateSwiftScriptCacheDirectory() throws -> AbsolutePath {
```

Comment:
Could this instead just call `getOrCreateSwiftPMCacheDirectory()` and then append the `scripts` subdirectory to it?  Otherwise this function seems to duplicate some of the existing code.

Reply:
Ummm... In fact the `scripts` dir is supposed to be beside `cache`, which I'm afraid may "escape" if we simply append `../scripts` to the cache dir (also, we need to create it if there's none, which needs a function to wrap up).
I thought it may be better to hardcode the path with `~/.swiftpm/scripts` to make it more accessible.